src/js/utils/lifecycle-callbacks.js

```diff
@@ -17,4 +17,15 @@ export default class LifecycleCallbacksMixin {
     }
     this.callbackQueues[queueName].push(callback);
   }
+  addCallbackOnce(queueName, callback) {
+    if (!queueName) { throw new Error('Must pass queue name to addCallbackOnce'); }
+    if (!this.callbackQueues[queueName]) {
+      this.callbackQueues[queueName] = [];
+    }
+    let index = this.callbackQueues[queueName].indexOf(callback);
+    if (index !== -1) {
+      this.callbackQueues[queueName].splice(index, 1);
+    }
+    this.callbackQueues[queueName].push(callback);
```

Comment:
I felt this was a completeness item, that is should always schedule as late as it was last added. But yeah, where I use it today there is only one thing in the queue regardless. Seem ok to simplify :+1: will do.